### PR TITLE
Change categories view to a recyclerview

### PR DESCRIPTION
**Pull request by martintoften**
Transformed the category view to a RecyclerView and changed the progress drawable

screenshot_1497000108

### CODE REVIEW

File: app/src/main/java/explorer/city/com/cityexplorer/view/adapter/CategoryAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import explorer.city.com.cityexplorer.model.ScoreCategory
+import explorer.city.com.cityexplorer.view.adapter.viewHolder.CategoryViewHolder
+import explorer.city.com.cityexplorer.view.listener.OnItemClickListener
+
+class CategoryAdapter: RecyclerView.Adapter<CategoryViewHolder>() {
+
+    private val categories by lazy { mutableListOf<ScoreCategory>() }
+
+    fun addItems(items: List<ScoreCategory>?) {
+        if (items == null) return
```
Comment (csdodd) [inline app/src/main/java/explorer/city/com/cityexplorer/view/adapter/CategoryAdapter.kt]: Do you need this null check?

File: app/src/main/java/explorer/city/com/cityexplorer/view/adapter/CategoryAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        categories.clear()
+        categories.addAll(items)
+        notifyDataSetChanged()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, p1: Int): CategoryViewHolder {
```
Comment (csdodd) [inline app/src/main/java/explorer/city/com/cityexplorer/view/adapter/CategoryAdapter.kt]: `p1` should be renamed to `pos` or `position`

Comment (csdodd) [reply app/src/main/java/explorer/city/com/cityexplorer/view/adapter/CategoryAdapter.kt]: Or maybe it's not a position -- either way, rename it to something other than `p1`

File: app/src/main/java/explorer/city/com/cityexplorer/view/custom/CategoriesView.kt [hunk 3]
```diff
@@ -29,28 +25,16 @@ class CategoriesView: LinearLayout {
...
+        initRecyclerView()
     }
 
-    fun setCategories(categories: List<ScoreCategory>?) {
-        categories?.forEach {
-            addCategoryView(it)
-        }
+    private fun initRecyclerView() {
+        layoutManager = LinearLayoutManager(context)
+        isNestedScrollingEnabled = false
```
Comment (csdodd) [inline app/src/main/java/explorer/city/com/cityexplorer/view/custom/CategoriesView.kt]: Might be neater to define this in the layout xml (if possible)

**Review by csdodd — APPROVED** [review]
Just small comments. Looks much better already 👏

File: app/src/main/java/explorer/city/com/cityexplorer/view/adapter/CategoryAdapter.kt (hunk 1 above)
Comment (martintoften) [reply app/src/main/java/explorer/city/com/cityexplorer/view/adapter/CategoryAdapter.kt]: nope, I can actually use 

```
items?.let { categories.addAll(it) }
```

**Pull request merged**